File: Java/gigasecond/src/main/java/Gigasecond.java
import java.time.LocalDate;
import java.time.LocalTime;
import java.time.LocalDateTime;

class Gigasecond {

    private static final long GIGASECOND = 1_000_000_000;
    private LocalDateTime birthDate;

    Gigasecond(LocalDateTime birthDateTime) {
        this.birthDate = birthDateTime;
    }

    Gigasecond(LocalDate birthDate) {
        this(birthDate.atStartOfDay());
    }

    LocalDateTime getDateTime() {
        return this.birthDate.plusSeconds(GIGASECOND);
    }

}

File: kotlin/gigasecond/src/main/kotlin/Gigasecond.kt
import java.time.LocalDate
import java.time.LocalDateTime

const val gigasecond = 1_000_000_000L

class Gigasecond(localDateTime: LocalDateTime) {

    val date: LocalDateTime = localDateTime.plusSeconds(gigasecond)

    constructor(localDate: LocalDate) : this(localDate.atStartOfDay())
}